lib/augmentConfig.js

```diff
@@ -370,4 +404,53 @@ function addProcessorFunctions(config) {
 	return config;
 }
 
-module.exports = { augmentConfigExtended, augmentConfigFull };
+/**
+ * @param {StylelintConfig} fullConfig
+ * @param {string} configDir
+ * @param {string} filePath
+ * @return {StylelintConfig}
+ */
+function applyOverrides(fullConfig, configDir, filePath) {
+	let { overrides, ...config } = fullConfig;
+
+	if (!overrides) {
+		return config;
+	}
+
+	if (!Array.isArray(overrides)) {
+		throw Error(
+			'The `overrides` configuration property should be an array, e.g. { "overrides": [{ "files": "*.css", "rules": {}}] }.',
+		);
+	}
+
+	for (const override of overrides) {
+		const { files, ...configOverrides } = override;
+
+		if (!files) {
+			throw Error(
+				'Every object in the `overrides` configuration property should have a `files` property with globs, e.g. { "overrides": [{ "files": "*.css", "rules": {}}] }.',
```

Comment:
[suggestion] Also here.

```suggestion
				'Every object in the `overrides` configuration property should have a `files` property with globs, e.g. { "overrides": [{ "files": "*.css", "rules": {} }] }.',
```